projects/vdk-core/plugins/vdk-test-utils/src/vdk/plugin/test_utils/util_plugins.py

```diff
@@ -77,35 +78,80 @@ def initialize_job(self, context: JobContext) -> None:
 
 
 class SqLite3MemoryConnection(ManagedConnectionBase):
-    def _before_cursor_execute(self, cur: ManagedCursor) -> None:
-        super()._before_cursor_execute(cur)
-
-    def __init__(self, history: List):
-        super().__init__(logging.getLogger(__name__))
-        self.history = history
+    def __init__(self):
+        super().__init__(logging.getLogger(__name__), None)
         self.db = SqLite3MemoryDb()
 
-    def _cursor_execute(self, cur: ManagedCursor, query: str) -> None:
-        self.history.append(query)
-        super()._cursor_execute(cur, query)
-
     def _connect(self) -> PEP249Connection:
         return self.db.new_connection()
 
 
+class ValidatedSqLite3MemoryDbPlugin:
```

Comment:
Why move them to vdk-test-utils? Are they necessary anywhere else ?

Reply:
My presumption was to place classes ValidatedSqLite3MemoryDbPlugin and SyntaxErrorRecoverySqLite3MemoryDbPlugin along with DecoratedSqLite3MemoryDbPlugin - each of those is a customised opinionated implementation hooked to the corresponding managed cursor handling phase (we have 3 of those).

So DecoratedSqLite3MemoryDbPlugin is tracking statements_history. It is also too specific, however it's used by two tests, instead of one.
ValidatedSqLite3MemoryDbPlugin and SyntaxErrorRecoverySqLite3MemoryDbPlugin are not used by more than one specific test (yet). So that is why I agree they could stay within the scope of the test itself. What do you think?